ssz/utils.py

```diff
@@ -72,60 +72,57 @@ def get_items_per_chunk(item_size: int) -> int:
         raise Exception("Invariant: unreachable")
 
 
-def pack(serialized_values: Sequence[bytes]) -> Tuple[Hash32, ...]:
-    if len(serialized_values) == 0:
-        return (EMPTY_CHUNK,)
-
-    item_size = len(serialized_values[0])
-    items_per_chunk = get_items_per_chunk(item_size)
-
-    number_of_items = len(serialized_values)
-    number_of_chunks = (number_of_items + (items_per_chunk - 1)) // items_per_chunk
-
-    chunk_partitions = partition(items_per_chunk, serialized_values, pad=b"")
-    chunks_unpadded = (b"".join(chunk_partition) for chunk_partition in chunk_partitions)
+def pad_zeros(value: bytes) -> bytes:
+    assert len(value) < CHUNK_SIZE
+    return bytes(value.ljust(CHUNK_SIZE, b"\x00"))
 
-    full_chunks = tuple(Hash32(chunk) for chunk in take(number_of_chunks - 1, chunks_unpadded))
-    last_chunk = first(chunks_unpadded)
-    if len(tuple(chunks_unpadded)) > 0:
-        raise Exception("Invariant: all chunks have been taken")
-
-    return full_chunks + (Hash32(last_chunk.ljust(CHUNK_SIZE, b"\x00")),)
-
-
-def pack_bytes(byte_string: bytes) -> Tuple[Hash32]:
-    size = len(byte_string)
-    if size == 0:
-        return (EMPTY_CHUNK,)
 
+def to_chuncks(packed_data) -> Tuple[bytes]:
+    size = len(packed_data)
     number_of_full_chunks = size // CHUNK_SIZE
     last_chunk_is_full = size % CHUNK_SIZE == 0
 
     full_chunks = tuple(
-        byte_string[chunk_index * CHUNK_SIZE:(chunk_index + 1) * CHUNK_SIZE]
+        packed_data[chunk_index * CHUNK_SIZE:(chunk_index + 1) * CHUNK_SIZE]
         for chunk_index in range(number_of_full_chunks)
     )
     if last_chunk_is_full:
         return full_chunks
     else:
-        last_chunk = byte_string[number_of_full_chunks * CHUNK_SIZE:].ljust(CHUNK_SIZE, b"\x00")
+        last_chunk = pad_zeros(packed_data[number_of_full_chunks * CHUNK_SIZE:])
         return full_chunks + (last_chunk,)
 
 
+def pack(serialized_values: Sequence[bytes]) -> Tuple[Hash32, ...]:
+    if len(serialized_values) == 0:
+        return (EMPTY_CHUNK,)
+
+    data = b''.join([value for value in serialized_values])
+    return to_chuncks(data)
+
+
+def pack_bytes(byte_string: bytes) -> Tuple[Hash32]:
```

Comment:
I think this type hint should be `Tuple[bytes, ...]` as it's a variable return size **and** it doesn't seems to return hashes.